python/cuml/test/test_kmeans.py

```diff
@@ -48,8 +51,15 @@ def get_data_consistency_test():
     return X, y
 
 
+@pytest.fixture
+def random_state():
+    random_state = random.randint(0, 1e6)
```

Comment:
This will run differently every time then, right? Even if it's pre-seeded, it would vary based on the order and number of tests run. That seems tricky for debugging, unless I'm missing some way it can be fixed in place.

Reply:
The pattern that I've seen for this in the past (I think scipy is particularly ardent about it) is that if you think something might behave differently for different random seeds, you choose the state randomly and then log what seed you used (as we do in the following lines). That way every run can potentially exercise a new random seed but you can always go back and debug the specific seed that caused an issue.

Do we generally like that pattern or is there a better way to approach this?

Reply:
Let's discuss live today... I think the big pain point here is that it can be very hard to repro failing runs and the occasional very rare subtle issue here can easily be dismissed as a flaky test or even missed by accident when tests happen to rerun.

Reply:
We didn't quite get to this specific question today, and I didn't want to prolong discussion any further. I think the basic question here is for those tests where we're passing in `random_state` as a parameter, does it really make sense to test multiple random seeds? And even if so, do we really want to test the same seeds every time?

My argument would be that we really have no way of choosing a number of seeds that is "good enough" to iterate through, whereas selecting a seed randomly at least gives us the possibility of finding an issue that we might not have hit with the fixed set of seeds we would otherwise have used. If we used a fixed set of seeds, we might just be hiding an issue entirely.

Even if we frequently dismiss something that is uncovered with a random seed as a "flaky test", it gives us the _possibility_ of noticing a problem and a good paper trail to go back through in order to find a reproducer.

That being said, if we don't like this approach, I'd probably advocate for converting all but one random seed in each of the tests to quality or stress params.

Reply:
I think the proposed approach is the better one for the time being(before looking more in depth into `hypothesis` or other options). We get coverage of random seeds as well as getting record of a seed that fails (do we actually print the seed in CI with the current `debug` level?), so I would favor keeping this change

Reply:
Yep, I just checked the CI output, and it's in there.